arch/arm/src/gd32f4/gd32f4xx_enet.c

```diff
@@ -1718,10 +1718,6 @@ static void gd32_receive(struct gd32_enet_mac_s *priv)
 
           if (priv->dev.d_len > 0)
             {
-              /* Update the Ethernet header with the correct MAC address */
-
-              arp_out(&priv->dev);
```

Comment:
let's move arp_out/neighbor_out prototype to  net/arp/arp.h and net/neighbor/neighbor.h

Reply:
Depends on https://github.com/apache/nuttx/pull/7701